pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -118,19 +118,20 @@ func (c *Controller) reconcileShootKey(key string) error {
 	}
 
 	seedName := shoot.Spec.Cloud.Seed
-	if seedName == nil {
-		return fmt.Errorf("shoot has not yet been scheduled on a seed")
+	if seedName == nil && shoot.DeletionTimestamp == nil {
+		return fmt.Errorf("shoot has not yet been scheduled to a seed yet")
```

Comment:
Instead of doing all these modifications: Can't you simply check here if the deletion time stamp is set, and if yes, remove the finalizer accordingly?